postRoutes.js

```diff
@@ -0,0 +1,103 @@
+const express = require("express");
+const router = express.Router();
+const db = require("./data/db.js");
+
+router.get("/", (request, response) => {
+  db.find()
+    .then(res => {
+      if (res) {
+        response.status(200).json({ success: true, res });
+      } else {
+        status(404).json({ success: false, message: "No posts available." });
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      response.status(500).json({ success: false, err });
+    });
+});
+
+router.get("/:id", (request, response) => {
+  const { id } = request.params;
+  db.findById(id)
+    .then(res => {
+      if (res) {
+        response.status(200).json({ success: true, res });
+      } else {
+        response
+          .status(404)
+          .json({ success: false, message: "Post does not exist." });
+      }
+    })
+    .catch(err => {
+      response.status(500).json({ success: false, err });
+    });
+});
+
+router.get("/:id/comments", (request, response) => {
+  const { id } = request.params;
+  db.findPostComments(id)
+    .then(res => {
+      response.status(200).json({ success: true, res });
+    })
+    .catch(err => {
+      response.status(500).json({ success: false, err });
+    });
+});
+
+router.post("/", (request, response) => {
+  const post = request.body;
+  db.insert(post)
```

Comment:
Before inserting, you would want to check that the post object has all of the required fields for the database.